src/cascadia/WindowsTerminal/AppHost.cpp

```diff
@@ -379,39 +381,59 @@ void AppHost::_HandleCreateWindow(const HWND hwnd, RECT proposedRect, LaunchMode
 
     // Get the size of a window we'd need to host that client rect. This will
     // add the titlebar space.
-    const auto nonClientSize = _window->GetTotalNonClientExclusiveSize(dpix);
-    adjustedWidth = islandWidth + nonClientSize.cx;
-    adjustedHeight = islandHeight + nonClientSize.cy;
-
-    const COORD dimensions{ Utils::ClampToShortMax(adjustedWidth, 1),
-                            Utils::ClampToShortMax(adjustedHeight, 1) };
+    const til::size nonClientSize = _window->GetTotalNonClientExclusiveSize(dpix);
+    adjustedWidth = islandWidth + nonClientSize.width<long>();
+    adjustedHeight = islandHeight + nonClientSize.height<long>();
+
+    til::size dimensions{ Utils::ClampToShortMax(adjustedWidth, 1),
+                          Utils::ClampToShortMax(adjustedHeight, 1) };
+
+    // Find nearest monitor for the position that we've actually settled on
+    HMONITOR hMonNearest = MonitorFromRect(&proposedRect, MONITOR_DEFAULTTONEAREST);
+    MONITORINFO nearestMonitorInfo;
+    nearestMonitorInfo.cbSize = sizeof(MONITORINFO);
+    // Get monitor dimensions:
+    GetMonitorInfo(hMonNearest, &nearestMonitorInfo);
+    const til::size desktopDimensions{ gsl::narrow<short>(nearestMonitorInfo.rcWork.right - nearestMonitorInfo.rcWork.left),
+                                       gsl::narrow<short>(nearestMonitorInfo.rcWork.bottom - nearestMonitorInfo.rcWork.top) };
 
+    til::point origin{ ::base::saturated_cast<ptrdiff_t>(proposedRect.left),
+                       ::base::saturated_cast<ptrdiff_t>(proposedRect.top) };
     if (centerOnLaunch)
     {
-        // Find nearest monitor for the position that we've actually settled on
-        HMONITOR hMonNearest = MonitorFromRect(&proposedRect, MONITOR_DEFAULTTONEAREST);
-        MONITORINFO nearestMonitorInfo;
-        nearestMonitorInfo.cbSize = sizeof(MONITORINFO);
-        // Get monitor dimensions:
-        GetMonitorInfo(hMonNearest, &nearestMonitorInfo);
-        const COORD desktopDimensions{ gsl::narrow<short>(nearestMonitorInfo.rcWork.right - nearestMonitorInfo.rcWork.left),
-                                       gsl::narrow<short>(nearestMonitorInfo.rcWork.bottom - nearestMonitorInfo.rcWork.top) };
         // Move our proposed location into the center of that specific monitor.
-        proposedRect.left = nearestMonitorInfo.rcWork.left +
-                            ((desktopDimensions.X / 2) - (dimensions.X / 2));
-        proposedRect.top = nearestMonitorInfo.rcWork.top +
-                           ((desktopDimensions.Y / 2) - (dimensions.Y / 2));
+        origin = til::point{
+            nearestMonitorInfo.rcWork.left + ((desktopDimensions.width() / 2) - (dimensions.width() / 2)),
+            nearestMonitorInfo.rcWork.top + ((desktopDimensions.height() / 2) - (dimensions.height() / 2))
+        };
+    }
+
+    if (_logic.IsQuakeWindow())

```

Comment:
nit: should we check for `IsQuakeWindow` first, then `else if (centerOnLaunch)`? It's minor but I guess we could save a bit of time with having to calculate the origin again.